EncodingStreamingHighAvailability/media-services-high-availability-shared/Services/MediaServiceInstanceHealthStorageService.cs

```diff
@@ -0,0 +1,210 @@
+﻿namespace media_services_high_availability_shared.Services
+{
+    using media_services_high_availability_shared.Helpers;
+    using media_services_high_availability_shared.Models;
+    using Microsoft.Azure.Cosmos.Table;
+    using Microsoft.Extensions.Logging;
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+
+    public class MediaServiceInstanceHealthStorageService : IMediaServiceInstanceHealthStorageService
+    {
+        private readonly CloudTable table;
+        private readonly ILogger logger;
+        private const int takeCount = 100;
+        private static DateTime minDateTimeForTableStorage = new DateTime(1900, 1, 1);
+
+        public MediaServiceInstanceHealthStorageService(CloudTable table, ILogger logger)
+        {
+            this.table = table ?? throw new ArgumentNullException(nameof(table));
+            this.logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        }
+
+        public async Task<MediaServiceInstanceHealthModel> CreateOrUpdateAsync(MediaServiceInstanceHealthModel mediaServiceInstanceHealthModel)
```

Comment:
This class has a lot of repeating code. I will look at using maybe generic methods or class to have one more abstraction layer for storage based on Tables. This will help JobStatusStorageService class as well.

Reply:
This was addressed in latest commit.